code/socialDistribution/views.py

```diff
@@ -373,18 +374,27 @@ def author(request, author_id):
     """
 
     curr_user = LocalAuthor.objects.get(user=request.user)
-    author = get_object_or_404(LocalAuthor, pk=author_id)
 
-    # TODO: Should become an API request since won't know if author is local/remote
+    try:
+        author = LocalAuthor.objects.get(pk=author_id)
+        author_type = LOCAL
+
+    except LocalAuthor.DoesNotExist:
+        author = get_object_or_404(Author, id=author_id)
+        author_type = REMOTE
+
+    if author_type == LOCAL:
+        if curr_user.has_friend(author):
+            posts = author.posts.listed().get_friend()  # get both friend and public posts
+        else:
+            posts = author.posts.listed().get_public()  # get public posts only
 
-    if curr_user.has_friend(author):
-        posts = author.posts.listed().get_friend()
     else:
-        posts = author.posts.listed().get_public()
+        posts = InboxPost.objects.filter(author=author.get_url_id())
```

Comment:
Let's make sure we only show public posts here. It's that or you find out if the current user is a friend and actually allowed to see the friends post. I would say take the simple option and only show public posts on this page.

(Right now other users on the server can see friends posts that were not meant for them)

Otherwise, this is a really great trick and should be enough for part 2. We can easily build on this to fetch posts later on if we want to.

Reply:
Made the changes
Thank you! ♥